### PR TITLE
Availabilities and Search😳

### PR DESCRIPTION
**Pull request by schooljack**

### CODE REVIEW

File: app/controllers/availabilities_controller.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
+class AvailabilitiesController < ApplicationController
+	def index
+		@availabilities = Availability.where(presenter: current_user.presenter).order('availabilities.start_time ASC')
```
Comment (schoolandrew) [inline app/controllers/availabilities_controller.rb]: It'd be great if you could abstract this query into the `Availability model` to clean up the controller

File: app/controllers/presenter_profiles_controller.rb [hunk 2]
```diff
@@ -7,6 +7,7 @@ class PresenterProfilesController < ApplicationController
   def show
     @presenter = find_presenter
     @user = @presenter.get_user
+    @availability = @presenter.availabilitys.all
```
Comment (schoolandrew) [inline app/controllers/presenter_profiles_controller.rb]: `@presenter.availabilitys.all ` can be refactored to `@presenter.availabilitys`

File: app/controllers/availabilities_controller.rb (hunk 1 above)
Comment (schooljack) [inline app/controllers/availabilities_controller.rb]: Model:
`def for_presenter(presenter)
  return Availability.where(presenter: current_user.presenter).order('availabilities.start_time ASC')
end`
Controller
` @availabilities = Availability.for_presenter(current_user.presenter)`

File: app/controllers/presenters_controller.rb [hunk 3]
```diff
@@ -9,4 +9,24 @@ def new
...
+  	@subject = Subject.find_by(name: params[:subjects][:name])
+  	unless current_user.presenter.subjects.include?(@subject)
+      current_user.presenter.subjects << @subject 
+    end
+  	redirect_to presenter_edit_subjects_path
+  end
+
+  def remove_subject
+    @subject = Subject.find(params[:subject])
+    @subject.presenters.delete(current_user.presenter)
```
Comment (schoolandrew) [inline app/controllers/presenters_controller.rb]: This code implies that it's deleting all 'presenters' of '@subject'. Is that right?

Comment (schooljack) [inline app/controllers/presenters_controller.rb]: REMOVE, not delete. 
Only an admin can delete a subject. `remove_subject` removes the subject from the presenter's subjects.

File: app/views/presenters/index.html.erb [hunk 4]
```diff
@@ -1 +1,7 @@
-<h1> Presenter Dashboard </h1>
\ No newline at end of file
+<h1> Presenter Dashboard </h1>
```
Comment (schoolandrew) [inline app/views/presenters/index.html.erb]: The presenter navigation has already been abstracted to partial '_presenter_nav.html.erb'

File: app/controllers/availabilities_controller.rb (hunk 1 above)
Comment (schoolandrew) [inline app/controllers/availabilities_controller.rb]: 'def self.for_presenter(presenter)'

File: app/views/presenters/index.html.erb (hunk 4 above)
Comment (schooljack) [inline app/views/presenters/index.html.erb]: Kept that in a merge because I thought it was removed in error.

**Pull request merged**